core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -837,6 +837,11 @@ public SystemSessionProperties(
                         "Minimum number of join build side rows required to use partitioned join lookup",
                         optimizerConfig.getJoinPartitionedBuildMinRowCount(),
                         value -> validateNonNegativeLongValue(value, JOIN_PARTITIONED_BUILD_MIN_ROW_COUNT),
+                        false),
+                booleanProperty(
+                        USE_EXACT_PARTITIONING,
+                        "When enabled this forces exchanges to occur unless the partitioning matches exactly",
```

Comment:
maybe be more verbose here: `When enabled this forces data repartitioning unless the partitioning of upstream stage matches exactly what downstream stage expects`?